internal/github/jwt.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2022 Namespace Labs Inc; All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+
+	"namespacelabs.dev/foundation/internal/auth"
+	"namespacelabs.dev/foundation/internal/fnerrors"
+)
+
+const (
+	userAgent = "actions/oidc-client"
+)
+
+func JWT(ctx context.Context, audience string) (string, error) {
+	idTokenURL := os.Getenv("ACTIONS_ID_TOKEN_REQUEST_URL")
+	if idTokenURL == "" {
+		return "", fnerrors.BadDataError("empty ID token request URL")
+	}
+	idToken := os.Getenv("ACTIONS_ID_TOKEN_REQUEST_TOKEN")
+	if idToken == "" {
+		return "", fnerrors.BadDataError("empty ID token request token")
+	}
+
+	if audience != "" {
+		idTokenURL += fmt.Sprintf("&audience=%s", url.QueryEscape(audience))
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "GET", idTokenURL, nil)
+	if err != nil {
+		return "", fnerrors.InvocationError("jwt", "failed to create HTTP request: %w", err)
+	}
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	req.Header.Add("User-Agent", userAgent)
+	req.Header.Add("Authorization", "Bearer "+idToken)
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return "", fnerrors.InvocationError("jwt", "failed to get github JWT: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		if resp.StatusCode == http.StatusUnauthorized {
+			return "", auth.ErrRelogin
```

Comment:
This error is not ideal here.

If we get and unauth error while fetching the Github JWT, running `ns login` would not help resolve it.

Reply:
Added a comment (I am just not sure if default one is OK, most likely we would need to set one)

Reply:
dropped this part and return a general error